styx-scheduler-service/src/test/java/com/spotify/styx/flyte/FlyteAdminClientRunnerTest.java

```diff
@@ -283,6 +283,23 @@ public void testEmitFlyteEventsExceptionHandling() throws IsClosedException {
                 Execution.WorkflowExecution.Phase.SUCCEEDED).build()).build(), runState));
   }
 
+  @Test
+  public void testUndefinedShouldNotThrowException() throws Exception {
+    WorkflowInstance workflowInstance = createWorkflowInstance();
+
+    when(flyteAdminClient.getExecution("flyte-test", "testing", "execution-name")).thenReturn(
+        ExecutionOuterClass.Execution
+            .newBuilder()
+            .setClosure(ExecutionOuterClass.ExecutionClosure.newBuilder()
+                .setPhase(Execution.WorkflowExecution.Phase.UNDEFINED).build())
+            .build());
+    when(runState.workflowInstance()).thenReturn(workflowInstance);
```

Comment:
Why is this needed?

Reply:
because the logline inside the emitFlyteEvents calls this method